components/sup/src/manager.rs

```diff
@@ -1256,13 +1256,24 @@ impl Manager {
             self.spec_dir
                 .specs()
                 .into_iter()
-                .filter(|spec| !existing_idents.contains(&spec.ident))
-                .flat_map(|spec| {
-                    Service::new(self.sys.clone(),
-                                 spec,
-                                 self.fs_cfg.clone(),
-                                 self.organization.as_ref().map(String::as_str),
-                                 self.state.gateway_state.clone()).into_iter()
+                .filter_map(|spec| {
+                    let ident = spec.ident.clone();
```

Comment:
Why not move the `clone` inside the `else` case so we can avoid it when unnecessary?